tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/Filer.java

```diff
@@ -40,100 +39,80 @@ public Filer setLanguagePlugin(LanguagePlugin languagePlugin) {
     public FileVisitResult decideOnDirectory(Path directory) {
         return FileVisitResult.CONTINUE;
     }
-
-    public List<String> prepareFile(List<String> file) {
-        return file;
-    }
-
-    List<CommentStyleFileFilter> getCommentStyleFileFilters() {
-        return new ImmutableList.Builder<CommentStyleFileFilter>()
-                .add(
-                        // Basic java
-                        new CommentStyleFileFilter(
-                                // JavaishCommentSyntax
-                                CommentSyntax.newBuilder()
-                                        .addSingleLineComment("\\/\\/")
-                                        .addMultiLineComment("\\/\\*+", "\\*+\\/")
-                                        .build()))
-                .add(
-                        new CommentStyleFileFilter(
-                                // XML comment syntax
-                                CommentSyntax.newBuilder()
-                                        .addMultiLineComment("/\\*", "\\*/")
-                                        .build()))
-                //TODO: add rest of the comment syntaxes
-                .build();
+    
+    public void visitFile(Path source, Path relativePath) {
+        logger.info("Looking into file: {} ", source);
+        String skipRegex = "\\.tmcrc|metadata\\.yml|(.*)Hidden(.*)";
+        if (source.getFileName().toString().matches(skipRegex)) {
+            logger.info("Skipping file: {} ", source);
+            return;
+        }
+        Path destination = toPath.resolve(relativePath);
+        try {
+            copyWithFilters(source, destination);
+        } catch (IOException ex) {
+            throw new IllegalStateException(ex);
```

Comment:
Since IOException is checked we don't want to flood everything with `throws IOException`. Thus I initially had it wrapped just in IllegalStateException.
Imo we should create something like `TmcLangsRuntimeException extends RuntimeException` which would be used to push the IOExceptions and others which we just want to crash for up the call stack and stop the command.